metrics/src/main/java/mobile/aerogear/org/metrics/MetricsService.java

```diff
@@ -0,0 +1,122 @@
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+import org.aerogear.mobile.core.logging.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.UUID;
+
+public class MetricsService implements ServiceModule {
+    private final static String MODULE_NAME = "metrics";
+    private final static String LOG_TAG = "AEROGEAR/METRICS";
+    private final static String STORAGE_NAME = "mobile.aerogear.org.metrics";
+    private final static String STORAGE_KEY = "metrics-sdk-installation-id";
+
+    private HttpServiceModule httpService;
+    private MetricsConfig config;
+    private Logger logger;
+
+    /**
+     * Get or create the client ID that identifies a device as long as the user doesn't
+     * reinstall the app or delete the app storage. A random UUID is created and stored in the
+     * application shared preferences.
+     *
+     * @param context Android app context
+     * @return String Client ID
+     */
+    private String getOrCreateClientId(final Context context) {
+        final SharedPreferences preferences = context
+            .getSharedPreferences(STORAGE_NAME, Context.MODE_PRIVATE);
+
+        String clientId = preferences.getString(STORAGE_KEY, null);
+        if (clientId == null) {
+            clientId = UUID.randomUUID().toString();
+
+            logger.info(LOG_TAG, "Generated a new client ID: " + clientId);
+
+            SharedPreferences.Editor editor = preferences.edit();
+            editor.putString(STORAGE_KEY, clientId);
+            editor.commit();
+        }
+
+        return clientId;
+    }
+
+    /**
+     * Get the version of the app itself
+     *
+     * @param context Android application context
+     * @return String version name
+     */
+    private String getAppVersion(final Context context) {
+        try {
+            return context
+                .getPackageManager()
+                .getPackageInfo(context.getPackageName(), 0).versionName;
+        } catch (PackageManager.NameNotFoundException e) {
+            logger.error(LOG_TAG, e);
+            return null;
+        }
+    }
+
+    private byte[] getMetricsData(final Context context) throws JSONException {
+        final JSONObject result = new JSONObject();
+        result.put("clientId", getOrCreateClientId(context));
+        result.put("appId", context.getPackageName());
+        result.put("appVersion", getAppVersion(context));
+        result.put("sdkVersion", BuildConfig.VERSION_NAME);
+
+        return result.toString().getBytes();
+    }
+
+    /**
+     * Send the metrics data to the server. The data is contained in a JSON object with the
+     * following properties: clientId, appId, sdkVersion and appVersion
+     *
+     * @param context Android application context
+     */
+    public void init(final Context context) {
```

Comment:
We should raise a RuntimeException here if we got some problem. I'd like to suggest to use `InitializationException`

Reply:
Hmm, not sure. If the metrics request fails for some reason, do we really want to crash the whole app? Maybe the the backend is fully functional and only the metrics server has been stopped. In that case the apps should still work.

Reply:
Makes sense

Reply:
IMO this depends on how we call this `init` method. Perhaps we can still throw the exception and let the caller decide what to do with it.
Looked for callers, couldn't find any at the moment. How is this supposed to be called?

Reply:
@aliok At the moment the user has to call this explicitely in their App. We discussed the option of making this implicit, but decided against it for now.

Reply:
I really don't have any strong opinion on this. I'd like to hear @wei-lee @wtrocki and @secondsun

Reply:
+1 for collecting metrics manually now. I will be more happy to change method name to something similar to IOS implementation. For example `collectStartupMetrics` etc.